src/Microsoft.OData.Edm/Csdl/Serialization/EdmModelCsdlSchemaJsonWriter.cs

```diff
@@ -0,0 +1,1292 @@
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+using Microsoft.OData.Edm.Vocabularies;
+using Microsoft.OData.Edm.Vocabularies.V1;
+
+namespace Microsoft.OData.Edm.Csdl.Serialization
+{
+    /// <summary>
+    /// OData Common Schema Definition Language (CSDL) JSON writer
+    /// </summary>
+    internal class EdmModelCsdlSchemaJsonWriter : EdmModelCsdlSchemaWriter
+    {
+        private Utf8JsonWriter jsonWriter;
+        private CsdlJsonWriterSettings settings;
+        private bool isInEnumTypeWriting;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EdmModelCsdlSchemaJsonWriter"/> class.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="writer">The JSON writer.</param>
+        /// <param name="edmVersion">The Edm version.</param>
+        internal EdmModelCsdlSchemaJsonWriter(IEdmModel model, Utf8JsonWriter writer, Version edmVersion)
+            : this(model, writer, edmVersion, CsdlJsonWriterSettings.Default)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="EdmModelCsdlSchemaJsonWriter"/> class.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="writer">The JSON writer.</param>
+        /// <param name="edmVersion">The Edm version.</param>
+        /// <param name="options">The CSDL serializer options.</param>
+        internal EdmModelCsdlSchemaJsonWriter(IEdmModel model, Utf8JsonWriter writer, Version edmVersion, CsdlJsonWriterSettings settings)
+            : base(model, edmVersion)
+        {
+            EdmUtil.CheckArgumentNull(writer, "writer");
+            EdmUtil.CheckArgumentNull(settings, "settings");
+
+            this.jsonWriter = writer;
+            this.settings = settings;
+        }
+
+        /// <summary>
+        /// Write Term Object header
+        /// </summary>
+        /// <param name="term">The Edm Term</param>
+        /// <param name="inlineType">Is inline type or not.</param>
+        internal override void WriteTermElementHeader(IEdmTerm term, bool inlineType)
+        {
+            // A term is represented as a member of the schema object whose name is the unqualified name of the term.
+            this.jsonWriter.WritePropertyName(term.Name);
+
+            // whose value is an object.
+            this.jsonWriter.WriteStartObject();
+
+            // The term object MUST contain the member $Kind with a string value of Term.
+            this.jsonWriter.WriteRequiredProperty("$Kind", CsdlConstants.Element_Term);
+
+            // It MAY contain the members $Type, $Collection.
+            if (inlineType && term.Type != null)
+            {
+                WriteTypeReference(term.Type);
+            }
+
+            // A term MAY specialize another term in scope by specifying it as its base term.
+            // The value of $BaseTerm is the qualified name of the base term. So far, it's not supported.

```

Comment:
>So far, it's not supported [](start = 78, length = 26)

Can you create an issue to track this as a (Pri3) work item?

Reply:
Done at [https://github.com/OData/odata.net/issues/1816](https://github.com/OData/odata.net/issues/1816)

---
In reply to: [443711595](https://github.com/OData/odata.net/pull/1775#discussion_r443711595) [](ancestors = 443711595)